### PR TITLE
#2485. Add is T constant evaluation tests

### PR DESCRIPTION
**Pull request by sgrekhov**

### CODE REVIEW

File: Language/Expressions/Constants/as_type_A01_t02.dart [hunk 1]
```diff
@@ -73,7 +73,8 @@ main() {
   const fo7 = IntET2(1) as FutureOr<IntET1Alias>;
 
   const f1 = foo as void Function();
-  const f2 = bar<int> as int Function<int>(int);
+  const f2 = as as void Function();
+  const f3 = bar<int> as int Function<int>(int);
```
Comment (eernstg) [inline Language/Expressions/Constants/as_type_A01_t02.dart:77]: If this is not intended to be a compile-time error I think we'd do this:
```suggestion
  const f3 = bar<int> as int Function(int);
```

File: Language/Expressions/Constants/is_type_A02_t02.dart [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+// [cfe] unspecified
+  const Void = void is Type;
+//             ^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  const cIntExtension = IntExtension is Type;
+//                      ^^^^^^^^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+  const F = () {} is Function;
```
Comment (eernstg) [inline Language/Expressions/Constants/is_type_A02_t02.dart:32]: This is an expression, just not a constant one. Fits in a different test?

**Review by eernstg — COMMENTED** [review]
Looks good! Just a couple of glitches.

File: Language/Expressions/Constants/as_type_A01_t02.dart (hunk 1 above)
Comment (sgrekhov) [reply Language/Expressions/Constants/as_type_A01_t02.dart]: Updated

File: Language/Expressions/Constants/is_type_A02_t02.dart (hunk 2 above)
Comment (sgrekhov) [reply Language/Expressions/Constants/is_type_A02_t02.dart:32]: Indeed. Moved to appropriate test. Thanx!

**Review by sgrekhov — COMMENTED** [review]
Thank you! Updated. Please take another look

**Review by eernstg — APPROVED** [review]
LGTM

**Pull request merged**